designs/Consolidate-NuGetConfig_OnLinux.md

```diff
@@ -0,0 +1,216 @@
+
+1.If the two config files are the same, NuGet will keep the one in consolidated path. 
+Shall we do the generalized comparison on the two xml files? That is, if the two functionally work the same, then they're same. So comments, space, key name differences will be ignored. It may need extra work compared with using string compare. Is this worth doing? 
+
+2.If one of them is the default NuGet.Config file, NuGet will ignore the default one and choose the other one. Will any scenarios break if we ignore the default NuGet.Config file?
+If one of them is the default NuGet.Config file(has nuget.org as the only source), the other one is not and it doesn't have nuget.org as one of it's sources.
+
+The above two should have covered most of the situations when the two config files both exist. 
+The 3rd situation is, customers use both dotnet CLI and mono, but have different contents of user wide NuGet.Config file other than default NuGet.Config file. That seems not reasonable. Is there any data we could use to prove this assumption? 
+
+**The symbolic link related .NET APIs for non-Windows platform are available in net6.0 for now, but we need to retarget some of our projects to net6.0.**
+
+There is a [Proposed API for symbolic links](https://github.com/dotnet/runtime/issues/24271)
+And the following APIs are available in net6.0:
+* [File.ResolveLinkTarget(String, Boolean) Method](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.resolvelinktarget?view=net-6.0)
+* [File.CreateSymbolicLink(String, String) Method](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.createsymboliclink?view=net-6.0)
+
+When retargeting some of our projects to net6.0, are we going to add net6.0 into the current list of TFMs, or replace some of the old TFMs?
+
+## Future Work
+Get rid of the legacy path completely. 
+
+## Open Questions
+
+**Which dotnet version are we going to bring in this change?**
+
+The version of .NET with this change will show warning to users who still have NuGet.Config in legacy path, encouraging them to move it to consolidated path.
+So it's good if more people could get this warning and get prepared to get rid of legacy path compleletly for future versions.
+For now, .NET 7 is the earlest possible version we could have this change.
+
+**Shall we insert this change into dotnet 6.x as servicing patch?**
+
+6.x is a LTS version. If 6.x doesn't have this change, it will still use the legacy path, we could not get rid of the legacy path completely if there are still people using 6.x.
+But at the same time, the bar for servicing patch should be high and for fixing security issues. 
+
+**When can we get rid of the legacy path completely?**
+
+This version: the .NET version with this change. It will show warning to users who still have NuGet.Config in legacy path, encouraging them to move it to consolidated path. If customers react to the warning, then they migrate to the consolidated path.
```

Comment:
depends on which solution is chosen. There are many scenarios where this spec proposes silently doing something, so no warning will be displayed. In fact, from the proposals, it looks like the only scenario where a warning will be displayed is when the legacy and consolodated nuget.config files both exist, and can't be automatically merged.